internal/kibana/connector_data_source.go

```diff
@@ -0,0 +1,57 @@
+package kibana
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func DataSourceConnector() *schema.Resource {
+	var connectorSchema = map[string]*schema.Schema{
+		"connector_id": {
+			Description: "A UUID v1 or v4 randomly generated ID.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"space_id": {
+			Description: "An identifier for the space. If space_id is not provided, the default space is used.",
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     "default",
+		},
+		"name": {
+			Description: "The name of the connector. While this name does not have to be unique, a distinctive name can help you identify a connector.",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+		"connector_type_id": {
+			Description: "The ID of the connector type, e.g. `.index`.",
+			Type:        schema.TypeString,
+			Optional:    true,
```

Comment:
```suggestion
			Computed:    true,
```

Reply:
Changed this so we can search for the connector type now

Reply:
Does it make sense to search using both name and connector type? IMO `name` should be enough.